src/chord.cpp

```diff
@@ -817,4 +818,61 @@ int Chord::ResetDrawing(FunctorParams *functorParams)
     return FUNCTOR_CONTINUE;
 }
 
+int Chord::JustifyY(FunctorParams *functorParams)
+{
+    JustifyYParams *params = vrv_params_cast<JustifyYParams *>(functorParams);
+    assert(params);
+    assert(params->m_justificationSum > 0);
+
+    // Check if chord spreads across several staves
+    std::list<Staff *> extremalStaves;
+    for (Note *note : { this->GetTopNote(), this->GetBottomNote() }) {
+        Layer *layer = NULL;
+        Staff *staff = note->GetCrossStaff(layer);
+        if (!staff) staff = vrv_cast<Staff *>(this->GetFirstAncestor(STAFF));
+        assert(staff);
+        extremalStaves.push_back(staff);
+    }
+    assert(extremalStaves.size() == 2);
+
+    const int topStaffN = extremalStaves.front()->GetN();
+    const int bottomStaffN = extremalStaves.back()->GetN();
+    if (topStaffN < bottomStaffN) {
+        // Now calculate the shift due to vertical justification of the involved staves.
+        Object *measure = this->GetFirstAncestor(MEASURE);
+        if (!measure) return FUNCTOR_CONTINUE;
+
+        ListOfObjects staves;
+        ClassIdComparison matchType(STAFF);
+        measure->FindAllDescendantByComparison(&staves, &matchType, 1);
+
+        int shift = 0;
+        for (Object *staff : staves) {
+            Staff *currentStaff = vrv_cast<Staff *>(staff);
```

Comment:
Can't you use `Chord::GetCrossStaffExtremes` and simply use the two staves returned by it instead of going through all the staves of the measure? This looks quite cumbersome to me. Or am I overlooking something?

Reply:
Actually I tried this, but `GetCrossStaffExtremes` requires that a cross staff AND a cross layer is set on the top/bottom note, otherwise it returns nothing. So it was not working. I don't know if it should be like this, but I did not dare to touch it and produce any regressions.